### ISSUE TITLE
[Urgent] If I would like to coloring task bar, how should I do??

### ISSUE THREAD

**Issue by hilfiger1014** [jsGanttImproved/jsgantt-improved#65]
Hi Expert,
I have a requirement is if the user extended the project time then I must change the color of the task bar. For example:
If the original color is blue --> the task start date: 2017-12-30; end date: 2018-01-10. Now the User extended the end day to 2018-01-20. How do I make the task bar into two colors.
2017-12-30 - 2018-01-10 : blue
2018-01-10 - 2018-01-20 : Red

![34430654-2b5f60b0-eca2-11e7-9c88-fa3a3e4e26ba](https://user-images.githubusercontent.com/17022215/34438668-635c80a6-ece3-11e7-81bc-6cb6cdc12771.png)

Thanks for your help.

**Comment by UgurAsaner** [comment]
+1

**Comment by hilfiger1014** [comment]
Hi Expert,
I appreciate that you both help me to deal with this issue... but I give it a try and list the result that is not what I want.

![2018-03-08_13-35-24](https://user-images.githubusercontent.com/17022215/37134970-f5771a5a-22d5-11e8-97ff-76cb5ef22d04.png)

So .... I don't know how should I deal with it at the moment.

**Comment by UgurAsaner** [comment]
My answer above was including a parentId that refers a nonexistent ganttTask, also I removed the dependency feature from child Tasks.
 
So this example should work :
`g.AddTaskItem(new JSGantt.TaskItem(10,  'Parent Task',      '',           '',          '',  '',       0, 'Someone',  40,  2, 0,  1,     '',      '',      '', g));`
 `g.AddTaskItem(new JSGantt.TaskItem(101, 'First Child',     '2014-03-06','2014-03-09', 'gtaskblue',    '',       0, 'Brian',    60,  0, 10, 1,     '',      '',      '', g));  `
`g.AddTaskItem(new JSGantt.TaskItem(102, 'Second Child',    '2014-03-10','2014-03-20', 'gtaskred',    '',       0, 'Brian',    0,   0, 10, 1,     '',   '',      '', g));`
`g.AddTaskItem(new JSGantt.TaskItem(103, 'Third Child',     '2014-03-21','2014-03-31', 'gtaskgreen',    '',       0, 'Brian',    0,   0, 10, 1,     '',   '',      '', g));`

**Comment by hilfiger1014** [comment]
Thanks all experts. I have sorted it out.

**Comment by hilfiger1014** [comment]
Sorry to bother all experts, I forgot I have another questions. I have sorted my problem out using UgurAsaner's  solution. But I just found another question is I must put another bar over the original Gantt Bar like the bellow image.

![2018-03-08_22-15-05](https://user-images.githubusercontent.com/17022215/37155662-c5c7b262-231e-11e8-973f-2fe52020e32d.png)

I have two bars:
one is the color : yellow + color : orange + color : blue --> I have solved it using UgurAsaner's  solution.
another is the black color means the Actual Schedule --> This is my another question. How should I solve it.
